Stripe/Payments/STPPaymentHandler.m

```diff
@@ -340,8 +342,17 @@ - (BOOL)_handleSetupIntentStatusForAction:(STPPaymentHandlerSetupIntentActionPar
            [action completeWithStatus:STPPaymentHandlerActionStatusFailed error:[self _errorForCode:STPPaymentHandlerIntentStatusErrorCode userInfo:@{@"STPSetupIntent": setupIntent.description}]];
         case STPSetupIntentStatusRequiresPaymentMethod:
             // If the user forgot to attach a PaymentMethod, they get an error before this point.
-            // If authentication fails, or the card is declined, the SetupIntent transitions to this state.
-            [action completeWithStatus:STPPaymentHandlerActionStatusFailed error:[self _errorForCode:STPPaymentHandlerNotAuthenticatedErrorCode userInfo:nil]];
+            // If confirmation fails (eg not authenticated, card declined) the SetupIntent transitions to this state.
+            if ([setupIntent.lastSetupError.code isEqualToString:@"setup_intent_authentication_failure"]) {
```

Comment:
This string should probalby live in a const declared in `STPSetupIntentLastSetupError`